### PR TITLE
Minor fixes for clCopyImage.

### PR DESCRIPTION
**Pull request by JohnKesapidesARM**
* Use  get_image_size() for size calculations. 
* Report Image size in MBs on failure
* Add clFinish to ensure the trailing unmap call completes
 
Signed-off-by: John Kesapides <john.kesapides@arm.com>

### CODE REVIEW

File: test_conformance/images/clCopyImage/test_copy_generic.cpp [hunk 1]
```diff
@@ -632,5 +620,16 @@ int test_copy_image_generic( cl_context context, cl_command_queue queue, image_d
...
 
+    /**
+     * Call clFinish() to ensure the unmap call completes.
+     */
+    error = clFinish(queue);
+    if (error != CL_SUCCESS)
+    {
+        log_error("ERROR: clFinish() failed to return CL_SUCCESS: %s\n",
+                  IGetErrorString(error));
+        return -1;
```
Comment (jlewis-austin) [inline test_conformance/images/clCopyImage/test_copy_generic.cpp:631]: Any reason not to return the error here?

File: test_conformance/images/clCopyImage/test_copy_generic.cpp [hunk 2]
```diff
@@ -632,5 +620,16 @@ int test_copy_image_generic( cl_context context, cl_command_queue queue, image_d
         return error;
     }
 
+    /**
+     * Call clFinish() to ensure the unmap call completes.
```
Comment (jlewis-austin) [inline test_conformance/images/clCopyImage/test_copy_generic.cpp:624]: Good catch! I don't see this comment style elsewhere in the file. I'd suggest
`// Ensure the unmap call completes.`

File: test_conformance/images/clCopyImage/test_copy_generic.cpp (hunk 1 above)
Comment (JohnKesapidesARM) [reply test_conformance/images/clCopyImage/test_copy_generic.cpp]: ah yes, thank you,  it should return the error here. Uploaded a fix.

File: test_conformance/images/clCopyImage/test_copy_generic.cpp (hunk 2 above)
Comment (JohnKesapidesARM) [reply test_conformance/images/clCopyImage/test_copy_generic.cpp]: Cheers. I updated it, to conform to style.

**Review by jlewis-austin — APPROVED** [review]
Thanks, LGTM

**Review by kpet — APPROVED** [review]

**Pull request merged**